### ISSUE TITLE
Execution Block ID uid://A001/X15a0/X16e Sgr_A_st_ai_03_7M

### ISSUE THREAD

**Issue by keflavich** [ACES-CMZ/reduction_ACES#143]
Sgr_A_st_ai_03_7M
[uid://A001/X15a0/X16e](https://almascience.org/aq/?result_view=observation&mous=uid://A001/X15a0/X16e)

* [x] Observations completed?
* [ ] Delivered?
* [ ] Downloaded? (specify where)
  * [ ] hipergator
  * [ ] hipergator pipeline run

* [ ] [Weblog](https://data.rc.ufl.edu/secure/adamginsburg/ACES/weblogs/humanreadable/Sgr_A_st_ai_03_7M/html/) unpacked
* [ ] [Weblog](https://data.rc.ufl.edu/secure/adamginsburg/ACES/weblogs/humanreadable/Sgr_A_st_ai_03_7M/html/) Quality Assessment?
* [ ] Imaging: Continuum
* [ ] Imaging: Lines

  * [ ] SPW16 H13CN
  * [ ] SPW18 H13CO+/SiO
  * [ ] SPW20 HCO+
  * [ ] SPW22 HNCO
  * [ ] SPW24 Cont 1
  * [ ] SPW26 Cont 2

**Comment by d-l-walker** [comment]
## Summary
Red warning about data flagged in one EB:
- QA 45.35% data in uid___A002_Xf934b1_X127c0.ms flagged by online, shadow, qa0, qa2, before and template flagging agents
- I don't think this is a major issue. It's only 1/6 EBs, and the final sensitivity looks OK.
- No significant issues otherwise prior to imaging stage.

## Cubes
SPW 24 diverged and will need to be recleaned using cyclefactor/cycniter parameter. @d-l-walker will do this.
![uid___A001_X15a0_X16e s38_0 Sgr_A_star_sci spw24 cube I iter1 image pbcor sky](https://user-images.githubusercontent.com/13015266/171448105-7a78e079-010b-49ff-ae2f-e37b2bb29ab5.png)

The continuum channel selection in SPW 20 is ... questionable. It clearly includes significant line emission, which is evident in the 'line-free' moment 8 map. Since I'll be reprocessing this data for SPW 24, I may as well re-do SPW 20 and the continuum, using a more suitable channel range for SPW 20.

![uid___A001_X15a0_X16e s38_0 Sgr_A_star_sci spw20 cube I iter1 image spectrum](https://user-images.githubusercontent.com/13015266/171448144-e2232950-a465-4cc0-84f4-5d390f12186d.png)

![uid___A001_X15a0_X16e s38_0 Sgr_A_star_sci spw20 cube I iter1 image mom8_fc sky](https://user-images.githubusercontent.com/13015266/171448163-63495734-c9b7-4a7e-a342-10b82450562d.png)

**Comment by d-l-walker** [comment]
I've finished the re-imaging and uploaded the products to `/upload/Sgr_A_st_ai_03_7M_reclean/` @keflavich 

There were only two real changes -- I simply re-ran the PL steps for imaging, with a refined cont.dat file to minimise the line contamination in the continuum selection for SPW 20. The continuum + SPWs 20 & 24 were imaged, and `cyclefactor=1.5` was used to avoid the divergence in SPW 24. All products look good, so I'm setting this as 'Done'.

**Comment by keflavich** [comment]
Todo:
 * [ ] add continuum selection `spw` command to `override_tclean_parameters.json`
 * [ ] add `cyclefactor` to same
 * [ ] add `cont.dat` to (new) `continuum_selection` folder

**Comment by d-l-walker** [comment]
@keflavich to address your comment ... 
- I actually didn't use the `spw` parameter to clean the continuum. I simply re-ran the PL imaging steps with the new `cont.dat` file in the directory and let the PL do the rest.
- I updated the cyclefactor params for SPWs 20 & 24 via PR #207. 
- I noticed that the `/~/upload/Sgr_A_st_ai_03_7M_reclean/` folder is still present, so I just added the new `cont.dat` file there. I'm not sure where the new `continuum_selection` folder is? So just thought I'd chuck it in the relevant upload folder for now.

**Comment by keflavich** [comment]
TODO: check whether cont.dat file is in the right place to be used if we re-run everything

**Comment by pyhsiehATalma** [comment]
@d-l-walker @keflavich 

Thanks for reclean spw 24. I downloaded spw 24 cube under /working for CS2-1 feathering and found the cube still shows divergence at ch 602. Perhaps the cube is not updated correctly after reclean?

![X15a0_X16e_ch602](https://user-images.githubusercontent.com/95929782/232445369-d789278a-f872-4958-87c8-3a630e8193cf.jpg)

**Comment by d-l-walker** [comment]
Hi @pyhsiehATalma,

Could you grab the cube from /rawdata/2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X15a0_X16e/Sgr_A_st_ai_03_7M_reclean/uid___A001_X1590_X30a8.s3_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean.image.fits

I think the one that you used was the old cube.

**Comment by d-l-walker** [comment]
I've confirmed that channel 602 looks OK in the recleaned version. Could you try again using the other cube @pyhsiehATalma?

![Screenshot 2023-05-11 at 15 00 00](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/5514d884-0b08-4c71-aae9-96c0389fbd33)

**Comment by pyhsiehATalma** [comment]
> Hi @pyhsiehATalma,
> 
> Could you grab the cube from `/rawdata/2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X15a0_X16e/Sgr_A_st_ai_03_7M_reclean/uid___A001_X1590_X30a8.s3_0.Sgr_A_star_sci.spw24.cube.I.iter1.reclean.image.fits`
> 
> I think the one that you used was the old cube.

Hi @d-l-walker thanks for the re-imaging of spw 24. The map looks pretty good.
I just found that the file names under /Sgr_A_st_ai_03_7M_reclean should be replaced to "...uid___A001_X15a0_X16e...."

**Comment by keflavich** [comment]
I linked all the files to calibrated/working/ now:
```
/orange/adamginsburg/ACES/rawdata/2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X15a0_X16e/calibrated/working$ ln -s ../../Sgr_A_st_ai_03_7M_reclean/* .
```

**Comment by keflavich** [comment]
Since you did the feathering - that means that, because the files weren't in the right directory, you feathered with the wrong file and that is the one that ended up in the uploads/ directory, right?  The mistake's on my end, I just want to make sure I understand where it propagated.  The path forward is then what you're doing - re-feathering & statconting, pushing to the same upload directory, and then re-running the big mosaic making.

**Comment by d-l-walker** [comment]
Yeah exactly. I set up all my Globus CLI scripts to pull the relevant cubes from all `~/../../calibrated/working/` directories for all SBs, so if the recleaned files were elsewhere they weren't picked up.

**Comment by d-l-walker** [comment]
I've re-made the 12m7mTP cube and replaced the existing cube in `/upload/Feather_12m_7m_TP/SPW33/cubes/`.

I'll also upload the fixed and statcont'd 7M cube shortly.

Here's a comparison with the previous cube, which had divergence coming from the 7M data

![Image](https://github.com/user-attachments/assets/96e9ec19-4de9-430c-bf58-057f41e67108)

**Comment by d-l-walker** [comment]
I've now added the re-cleaned and statcont'd 7M SPW24 cube to `/upload/Sgr_A_st_ai_03_7M_SPW24_reclean/`

`uid___A001_X15a0_X16e.s12_0.Sgr_A_star_sci.spw24.cube.I.iter1.image.pbcor.statcont.contsub.fits` is the relevant cube

I also added all of the tclean output files for completeness, but note that they have a slightly different naming convention (e.g. `.reclean` in filenames)
